### ISSUE TITLE
[Bug]In Firestore for "Get Documents in Collection" method the "where condition" doesn't work for Number, Null, Boolean, Array of Numbers, Geolocation

### ISSUE THREAD

**Issue by RakshaKShetty** [appsmithorg/appsmith#6060]
## Description In Firestore for "Get Documents in Collection" method the "where condition" doesn't work for Number, Null, Boolean, Array of Numbers, Geolocation
[What happened]

https://loom.com/share/0665bef56ad14ceb9a3b00bbeaf8594a

### Steps to reproduce the behaviour:
1.Add the new data source Firestore
2.Select method  as "get documents in collection"
3.Add document/collection path
4. under where the condition 
5. add the path and the value should be a "number" data type 
6. then the output will be displayed as empty array
![image](https://user-images.githubusercontent.com/45958978/126508197-aae04f6d-e343-4ff9-aafc-51b150090444.png)

### Important Details

- Version: [Cloud ]
- OS: [MacOSX]
- Browser [ safari]
- Environment [release frozen]

**Comment by ajinkyakulkarni** [comment]
@trishaanand Hey team! Can you please add your [planning poker estimates with ZenHub](https://app.zenhub.com/workspaces/Appsmith-Actions-Pod-603dd87bd4dff90011242dc9/issues/appsmithorg/appsmith/6060?planning-poker)

**Comment by ajinkyakulkarni** [comment]
@trishaanand @ohansFavour Hey team! Can you please add your [planning poker estimates with ZenHub](https://app.zenhub.com/workspaces/Appsmith-Actions-Pod-603dd87bd4dff90011242dc9/issues/appsmithorg/appsmith/6060?planning-poker)

**Comment by trishaanand** [comment]
We can infer the data types and parse them accordingly before actually executing the query, but there is no way for the user to override these data types inferred.

**Comment by areyabhishek** [comment]
A user reported this bug on Intercom again today. See message below:
"also, i had reported this bug on discord like 3 months ago, where if you try to make firestore query with ==, <=, >= and a number, it just returns empty array where it should return proper docs. I just checked and it's still not fixed. It basically makes firestore query unusable if you want to use a number in the query."



@trishaanand @ohansFavour I can see that this issue was assigned.

**Comment by sumitsum** [comment]
Removing the following special data types from the scope since we would need to have few design related discussions before we can proceed with implemention: (1) GeoPoint (2) Reference (3) Timesamp. 
We are tracking the support for special data types here: https://github.com/appsmithorg/appsmith/issues/3741

**Comment by leotom2000** [comment]
Number - works
==, <, <=, >, >=

IN - Works
[]

Boolean - Works
==

Array 
- Backend is implemented , there "ARRAY_CONTAINS_ANY"  "ARRAY_CONTAINS"
- Frontend not implemented, Need "contains any"  "contains"
Need to create a FE issue

Geolocation
-  Not implemented
Need to create FE and BE issue

**Comment by RakshaKShetty** [comment]
Still facing the issue 
https://www.loom.com/share/f9ab0c152e5943ce802274a5a8c8cc45

**Comment by sumitsum** [comment]
@leotom2000 can you please check out the video Raksha has shared ?

**Comment by leotom2000** [comment]
@ajinkyakulkarni  @rohan-arthur 
We need a functional clarification regarding this reopened issue.

Firestore offers two operators "array-contains" and "array-contains-any"
The QA has tested it like a string, even in future end-users may do the same.
The reason in we are mentioning in Where condition as "Contains"
![image](https://user-images.githubusercontent.com/8027694/168530943-c2a5b8a6-0296-40d1-8f6a-680fcd1800b5.png)

Here the implementation is correct, but the way the general user may search is unguided.
The expectation here is the user should search in an array.

(1) Should we change the dropdown to "Array Contains", "Array Contains Any" in the dropdown.
The draw back is the field will be wider and inconsistent, 

(2) Should we retain this and throw a error message informing the user to search in an array.

Here is the docs related to Query operator
https://firebase.google.com/docs/firestore/query-data/queries

![image](https://user-images.githubusercontent.com/8027694/168533702-2e211432-8e20-412a-8fd5-e7967b78990d.png)

**Comment by leotom2000** [comment]
Closing this issue as there are no changes in the implementation.
A UI change is required to resolve the misunderstanding and that will be addressed by another issue below
https://github.com/appsmithorg/appsmith/issues/13880
